source/lib/index.ts

```diff
@@ -75,8 +78,15 @@ export default async (options: Options = {cwd: process.cwd()}) => {
 		config
 	};
 
-	return [
-		...getCustomDiagnostics(context),
-		...getTSDiagnostics(context)
-	];
+	const tsDiagnostics = getTSDiagnostics(context);
+	const customDiagnostics = getCustomDiagnostics(context);
+	const numTests = tsDiagnostics.numTests + customDiagnostics.numTests;
```

Comment:
Looks like the incorrect number of tests are coming from here. This is has to be further looked into.